crates/pop-contracts/src/init_tests.rs

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::{create_smart_contract, Contract};
+use anyhow::Result;
+use std::{fs, path::PathBuf};
+
+pub fn generate_smart_contract_test_environment() -> Result<tempfile::TempDir> {
+	let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+	let temp_contract_dir = temp_dir.path().join("testing");
+	fs::create_dir(&temp_contract_dir)?;
+	create_smart_contract("testing", temp_contract_dir.as_path(), &Contract::Standard)?;
+	Ok(temp_dir)
+}
+
+// Function that mocks the build process generating the contract artifacts.
+pub fn mock_build_process(
+	temp_contract_dir: PathBuf,
```

Comment:
"In public interfaces, you usually don’t want to use Path or PathBuf directly, but rather a generic P: AsRef<Path> or P: Into<PathBuf>. That way the caller can pass in Path, PathBuf, &str or String."

https://nick.groenen.me/notes/rust-path-vs-pathbuf/

Reply:
Nice. Is a better approach  to use a single generic parameter
```rs
pub fn mock_build_process<P>(temp_contract_dir: P, contract_file: P, metadata_file: P) -> Result<()>
where
	P: AsRef<Path>,
{
```
or one for each?
```rs
pub fn mock_build_process<P1, P2, P3>(
    temp_contract_dir: P1,
    contract_file: P2,
    metadata_file: P3,
) -> Result<()> 
where
    P1: AsRef<Path>,
    P2: AsRef<Path>,
    P3: AsRef<Path>,
{
```

Reply:
I'll keep it simple for now with one.

Reply:
One, but you can probably also use `impl AsRef<Path>` to avoid the need for an explicit generic.